src/tokenizer.rs

```diff
@@ -0,0 +1,510 @@
+        || t == TokenType::CloseParenthesis.into()
+        || t == TokenType::OpenSquare.into()
+        || t == TokenType::CloseSquare.into()
+        || t == TokenType::SingleQuote.into()
+        || t == TokenType::DoubleQuote.into()
+        || t == TokenType::Plus.into()
+        || t == TokenType::Pipe.into()
+        || t == TokenType::Tilde.into()
+        || t == TokenType::Greater.into()
+        || t == TokenType::Equals.into()
+        || t == TokenType::Dollar.into()
+        || t == TokenType::Caret.into()
+        || t == TokenType::Slash.into()
+    // others all considered not word delimiters.
+}
+
+#[inline(always)]
+pub(crate) const fn is_hex(ch: u8) -> bool {
+    match ch {
+        b'0'..=b'9' => true,
+        b'a'..=b'f' => true,
+        b'A'..=b'F' => true,
+        _ => false,
+    }
+}
+
+#[inline(always)]
+pub(crate) fn is_whitespace(ch: u8) -> bool {
+    ch == TokenType::Space.into()
+        || ch == TokenType::Tab.into()
+        || ch == TokenType::Newline.into()
+        || ch == TokenType::Cr.into()
+        || ch == TokenType::Feed.into()
+}
+
+#[inline(always)]
+pub(crate) const fn is_quote(ch: u8) -> bool {
+    match ch {
+        b'\'' | b'"' => true,
+        _ => false,
+    }
+}
+
+///
+/// Returns the last index of the escape seq.
+///
+fn consume_escape(css: &[u8], start: usize) -> usize {
+    let mut next = start;
+    let char_codes = css;
+    assert!(char_codes.len() > next + 1);
+    let mut code = char_codes[next + 1];
+    if is_unescapable(code.try_into().unwrap()) {
+        // just consume the escape char
+    } else if is_hex(code) {
+        let mut hex_digits = 0;
+        // consume up to 6 hex chars
+        loop {
+            next += 1;
+            hex_digits += 1;
+            code = char_codes[next + 1];
+            if !(is_hex(code) && hex_digits < 6) {
+                break;
+            }
+        }
+
+        // if fewer than 6 hex chars, a trailing space ends the escape
+        if hex_digits < 6 && code == TokenType::Space.into() {
+            next += 1;
+        }
+    } else {
+        // the next char is part of the current word.
+        next += 1;
+    }
+
+    next
+}
+
+///
+/// Returns the last index of the bar css word.
+///
+fn consume_word(css: &[u8], start: usize) -> usize {
+    let mut next = start;
+    let char_codes = css;
+
+    loop {
+        let code = char_codes[next];
+        if is_word_delimiter(code) {
+            return next - 1;
+        } else if code == TokenType::Backslash.into() {
+            next = consume_escape(css, next) + 1;
+        } else {
+            // All other characters are part of the word
+            next += 1;
+        }
+        if next >= char_codes.len() {
+            break;
+        }
+    }
+
+    next - 1
+}
+
+pub(crate) type Range = (usize, usize);
+
+pub struct Token {
+    pub r#type: TokenType,
+    pub line: Range,
+    pub col: Range,
+    pub pos: Range,
+}
+
+impl ToString for Token {
```

Comment:
I think it would be better to implement `std::fmt::Debug` instead , so you could just print it without call `to_string`